src/com/linkedin/parseq/internal/SerialExecutor.java

```diff
@@ -69,24 +70,40 @@
   private final Executor _executor;
   private final RejectedSerialExecutionHandler _rejectionHandler;
   private final ExecutorLoop _executorLoop = new ExecutorLoop();
-  private final FIFOPriorityQueue<Runnable> _queue = new FIFOPriorityQueue<Runnable>();
+  private final TaskQueue<PrioritizableRunnable> _queue;
   private final AtomicInteger _pendingCount = new AtomicInteger();
   private final DeactivationListener _deactivationListener;
 
   public SerialExecutor(final Executor executor,
       final RejectedSerialExecutionHandler rejectionHandler,
-      final DeactivationListener deactivationListener) {
+      final DeactivationListener deactivationListener,
+      final TaskQueue<PrioritizableRunnable> taskQueue) {
     ArgumentUtil.requireNotNull(executor, "executor");
     ArgumentUtil.requireNotNull(rejectionHandler, "rejectionHandler" );
     ArgumentUtil.requireNotNull(deactivationListener, "deactivationListener" );
 
     _executor = executor;
     _rejectionHandler = rejectionHandler;
+    _queue = taskQueue;
     _deactivationListener = deactivationListener;
   }
 
   public void execute(final Runnable runnable) {
-    _queue.add(runnable);
+    if (runnable instanceof Prioritizable) {
+      _queue.add((PrioritizableRunnable)runnable);
+    } else {
+      _queue.add(new PrioritizableRunnable() {
```

Comment:
Good point. Can you drop "implements Executor"? I think this will clean up the implementation a bit.